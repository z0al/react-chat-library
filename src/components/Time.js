// @ts-check

// Packages
import React from 'react';
import { format } from 'date-fns';

/**
 * @typedef Props
 * @property {string} children
 * @property {string} [position]
 * @property {string} [format]
 * @property {object} [style]
 * @param {Props} props
 */
const Time = props => (
	<span className={props.position} style={props.style}>
		{format(props.children, props.format)}
		<style jsx>{`
			span {
				display: block;
				font-size: 0.65em;
				margin-top: 0.5em;
				text-align: right;
			}

			span.left {
				color: #aaa;
			}

			span.right {
				color: white;
			}
		`}</style>
	</span>
);

/** @type Props */
Time.defaultProps = {
	children: '',
	position: 'left',
	format: 'hh:mm A',
	style: {}
};

export default Time;
