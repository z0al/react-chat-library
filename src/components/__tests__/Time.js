import React from 'react';
import renderer from 'react-test-renderer';

import Time from '../Time';

test('should match the snapshot', () => {
	const tree = renderer.create(<Time />).toJSON();
	expect(tree).toMatchSnapshot();
});
