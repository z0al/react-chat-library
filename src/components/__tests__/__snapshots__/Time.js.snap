// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<span
  className="jsx-2099417310 left"
>
  Invalid Date
</span>
`;
